experimental/wazerotest/wazerotest.go

```diff
@@ -0,0 +1,605 @@
+	m.exportedMemoryDefinitions = make(map[string]api.MemoryDefinition)
+
+	for index, function := range m.Functions {
+		for _, exportName := range function.ExportNames {
+			m.exportedFunctions[exportName] = function
+			m.exportedFunctionDefinitions[exportName] = function.Definition()
+		}
+		function.module = m
+		function.index = index
+	}
+
+	for _, global := range m.Globals {
+		for _, exportName := range global.ExportNames {
+			m.exportedGlobals[exportName] = global
+		}
+	}
+
+	if m.ExportMemory != nil {
+		m.ExportMemory.module = m
+		m.exportedMemoryDefinitions["memory"] = m.ExportMemory.Definition()
+	}
+}
+
+// Global is an implementation of the api.Global interface, it represents a
+// global in a WebAssembly module.
+type Global struct {
+	internalapi.WazeroOnlyType
+
+	// Type of the global value, used to interpret bits of the Value field.
+	ValueType api.ValueType
+
+	// Value of the global packed in a 64 bits field.
+	Value uint64
+
+	// List of names that the globla is exported as.
+	ExportNames []string
+}
+
+func (g *Global) String() string {
+	switch g.ValueType {
+	case api.ValueTypeI32:
+		return strconv.FormatInt(int64(api.DecodeI32(g.Value)), 10)
+	case api.ValueTypeI64:
+		return strconv.FormatInt(int64(g.Value), 10)
+	case api.ValueTypeF32:
+		return strconv.FormatFloat(float64(api.DecodeF32(g.Value)), 'g', -1, 32)
+	case api.ValueTypeF64:
+		return strconv.FormatFloat(api.DecodeF64(g.Value), 'g', -1, 64)
+	default:
+		return "0x" + strconv.FormatUint(g.Value, 16)
+	}
+}
+
+func (g *Global) Type() api.ValueType {
+	return g.ValueType
+}
+
+func (g *Global) Get() uint64 {
+	return g.Value
+}
+
+func GlobalI32(value int32, export ...string) *Global {
+	return &Global{ValueType: api.ValueTypeI32, Value: api.EncodeI32(value), ExportNames: export}
+}
+
+func GlobalI64(value int64, export ...string) *Global {
+	return &Global{ValueType: api.ValueTypeI64, Value: api.EncodeI64(value), ExportNames: export}
+}
+
+func GlobalF32(value float32, export ...string) *Global {
+	return &Global{ValueType: api.ValueTypeF32, Value: api.EncodeF32(value), ExportNames: export}
+}
+
+func GlobalF64(value float64, export ...string) *Global {
+	return &Global{ValueType: api.ValueTypeF64, Value: api.EncodeF64(value), ExportNames: export}
+}
+
+// Function is an implementation of the api.Function interface, it represents
+// a function in a WebAssembly module.
+//
+// Until accessed through a Module's method, the function definition's
+// ModuleName method returns an empty string and its Index method returns 0.
+type Function struct {
+	internalapi.WazeroOnlyType
+
+	// GoModuleFunction may be set to a non-nil value to allow calling of the
+	// function via Call or CallWithStack.
+	//
+	// It is the user's responsibility to ensure that the signature of this
+	// implementation matches the ParamTypes and ResultTypes fields.
+	GoModuleFunction api.GoModuleFunction
+
+	// Type lists representing the function signature. Those fields should be
```

Comment:
is it common to doc above multiple fields? If so, teach me. If just a shortcut for now, also ok

Reply:
Can't say it is. But Go is pretty informal about this things. You can easily find a mix and match style of styles even in the same struct.
https://pkg.go.dev/net/http#Cookie
https://pkg.go.dev/database/sql#DBStats

Reply:
I have seen it done across the stdlib, godoc shows the source code of struct declarations so we have a bit of liberty as to how we layout the documentation.

@codefromthecrypt I can change it if you think the layout hurts readability.